### ISSUE TITLE
bpf: complexity-tests: split kernel capabilities into separate file

### ISSUE THREAD

**Issue by julianwiedmann** [cilium/cilium#32161]
Part of the BPF complexity test configuration is to spell out the capabilities of the tested kernel. For example https://github.com/cilium/cilium/blob/6092fca685dae8981e12b92423cf4af743810309/bpf/complexity-tests/netnext/bpf_host/1.txt#L1-L2

But for a given kernel version, these capabilities are identical across **all tested configurations** and **all programs**. Thus it would be much less maintenance effort to have a single `capabilities.txt` file in each kernel directory.

And then have https://github.com/cilium/cilium/blob/main/test/verifier/verifier_test.go mix the file content into the test configuration.

**Comment by avi028** [comment]
Hi @julianwiedmann  I would like to solve this issue.

**Comment by julianwiedmann** [comment]
> Hi @julianwiedmann I would like to solve this issue. Can you assign this to me

Absolutely, thank you! Please don't hesitate to ask for help (here or on Slack) when you've had time to dig in.

Note that you can test your changes locally doing something like
> go test -exec sudo -v -run TestVerifier/bpf_host/1 ./test/verifier/ -cilium-base-path $(pwd) -ci-kernel-version netnext

**Comment by yogesh1801** [comment]
@julianwiedmann is this issue still open to contribute? i would like to try it

**Comment by julianwiedmann** [comment]
@avi028 :wave: Hi! Are you making progress on this issue, or can @yogesh1801 take over?

**Comment by yogesh1801** [comment]
@julianwiedmann i will start working on this, if avi gives successfull PR then good else if will raise my PR, does that work?

**Comment by julianwiedmann** [comment]
> @julianwiedmann i will start working on this, if avi gives successfull PR then good else if will raise my PR, does that work?

sure, works for me :)

**Comment by umesh3034** [comment]
Need to consolidate all bpf_* capabilities per kernel ?

.
   |-510
   |---bpf_host
   |---bpf_lxc
   |---bpf_network
   |---bpf_overlay
   |---bpf_sock
   |---bpf_wireguard
   |---bpf_xdp
   |-54
   |---bpf_host
   |---bpf_lxc
   |---bpf_network
   |---bpf_overlay
   |---bpf_sock
   |---bpf_wireguard
   |---bpf_xdp
   |-61
   |---bpf_host
   |---bpf_lxc
   |---bpf_network
   |---bpf_overlay
   |---bpf_sock
   |---bpf_wireguard
   |---bpf_xdp
   |-netnext
   |---bpf_host
   |---bpf_lxc
   |---bpf_network
   |---bpf_overlay
   |---bpf_sock
   |---bpf_wireguard
   |---bpf_xdp

**Comment by julianwiedmann** [comment]
With the new complexity test suite these capabilities should be handled by running the actual BPF probes on the tested kernel.
